### ISSUE TITLE
Support Symfony 4

### ISSUE THREAD

**Issue by harentius** [harentius/blog-bundle#6]

**Comment by tacman** [comment]
I attempted to update a fork to Symfony 4, mostly by hacking at composer.json until it would at least install.

But now there's an issue registering the Sonata Twig extensions, not sure how to fix it.

My fork is located at  https://github.com/survos/blog-bundle, so far I've only changed the composer.json file, and the error I now get when clearing the cache is

```
Argument 3 passed to Sonata\Twig\Extension\TemplateExtension::__construct()  
!!     must be an instance of Sonata\CoreBundle\Model\Adapter\AdapterInterface, i  
!!    nstance of Sonata\Doctrine\Adapter\AdapterChain given, called in /home/tac/  
!!    github/transcribe/var/cache/dev/ContainerHBFIK7p/srcApp_KernelDevDebugConta  
!!    iner.php on line 1629 
```

Happy to help try to get this working with Symfony 4.

**Comment by harentius** [comment]
Hi @tacman 
Thank you so much for involving.
This is a quite complex task to clearly upgrade Symfony version from 3 to 4.
Regarding error you described: look like, some of versions to which you are trying to upgrade are not agreed. (Looks like problem with sonata dependencies)
I started to cover functionality with behavioral tests to simplify future version upgrades, but then abandon it.)
Are you going to use this bundle somewhere so trying to upgrade Symfony for it?

**Comment by tacman** [comment]
Eh, if it were easy to integrate, I was considering using it on a site I'm
working on, but it's not urgent at all.  Spent more time than I expected
just trying to get a set of dependencies to install.

We used to use Sonata AdminBundle, I see its value but indeed it's a lot of
dependencies.

BTW, I ran into an error installing the demo program, because the _locale
was set to 'en' and it only accepts 'ru', I should report that in that
repo, I guess.

Anyway, if you're able to get it to work in Symfony4, I would use it.

Tac
